### ISSUE TITLE
Mocked lambda returning a blank object when run inside a python container.

### ISSUE THREAD

**Issue by nrjchnd2** [spulec/moto#4073]
I have a use case where I am mocking a lambda function using moto. All works just fine on my local machine but as soon as I push the code and it runs inside a python:3.8-slim container (containerized git lab runner), somehow the mocked lambda is not returning proper response.

```
def _process_lambda(func_str):
    zip_output = io.BytesIO()
    zip_file = zipfile.ZipFile(zip_output, "w", zipfile.ZIP_DEFLATED)
    zip_file.writestr("lambda_function.py", func_str)
    zip_file.close()
    zip_output.seek(0)
    return zip_output.read()
    
    
def get_test_zip_file1():
        pfunc = """
    from datetime import datetime
    data={}
    data['execution'] = {
                'started_timestamp': datetime.utcnow().isoformat()
                }
    def lambda_handler(event, context):
        print("custom log event")
        return {'body':metadata}
    """
        return _process_lambda(pfunc)
```

And below is where I actually create the function and write some asserts.

```
@mock_lambda
def test_mock_some_lambda(lambda_mock):
    import app
    lambda_mock.create_function(
        FunctionName='abc',
        Runtime='python3.7',
        Role=get_role_name(),
        Handler='lambda_function.lambda_handler',
        Code={"ZipFile": get_test_zip_file1()},
        Publish=True,
        Timeout=30,
        MemorySize=128
    )
    response=lambda_mock.invoke(
        FunctionName='abc'
    )
    response=json.load(response['Payload'])
    assert 'body' in response
```

This passes just fine on my local but inside the container I am getting: AssertionError: assert 'body' in {}. So looks like the mock lambda does get created inside the container (otherwise invoke would have throws 404), but it's returning a {} instead of a dict having 'body' as one key.

How are these things behaving differently ?

**Comment by bblommers** [comment]
Hi @nrjchnd2, is Docker installed in that container? Moto spins up another Docker-container to handle the invocation, so if that's not installed, that might be the issue.

The Lambda-logs will give you more insight into what's going on:
```
response=lambda_mock.invoke(
        FunctionName='abc',
        LogType="Tail",
    )
logs = base64.b64decode(success_result["LogResult"]).decode("utf-8")
````

**Comment by nrjchnd2** [comment]
@bblommers  i was able to figure this out and resolve. So there were two issues.

1) I had to explicitly give InvocationType parameter while invoking the lambda. I though it would be able to pick up the default but that wasn't the case. So once I specified that parameter, it stopped sending {} response.

 2)I was using an old version of moto (1.3.14) which was sending messed up Payload from mocked lambda.
Once I upgraded the moto version to 2.0.5 , both these issues were sorted out.
